stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/security/KeyStore.java

```diff
@@ -0,0 +1,256 @@
+     * @throws KeyStoreException if the keystore has not been initialized
+     *                           (loaded), if the algorithm for recovering the entry cannot be found,
+     *                           {@code alias} or {@code keyPassword} does not contain
+     *                           the information needed to recover the key (e.g. wrong password)
+     */
+    public PrivateKey getPrivateKey(String alias, char[] keyPassword) throws KeyStoreException {
+        PrivateKeyEntry pkEntry;
+        try {
+            pkEntry = (PrivateKeyEntry) keyStore.getEntry(alias, new PasswordProtection(keyPassword));
+        } catch (NoSuchAlgorithmException | UnrecoverableEntryException | java.security.KeyStoreException e) {
+            throw new KeyStoreException("Failed to load private key: " + alias, e);
+        }
+        if (pkEntry == null) {
+            return null;
+        }
+        return pkEntry.getPrivateKey();
+    }
+
+    /**
+     * Get the public key for a given key alias from keyStore.
+     *
+     * @param alias the alias name
+     * @return public key corresponding to the alias
+     * @throws KeyStoreException if the keystore has not been initialized
+     *                           (loaded), or this operation fails for some other reason
+     */
+    public PublicKey getPublicKey(String alias) throws KeyStoreException {
+        Certificate certificate = getCertificate(alias);
+        if (certificate == null) {
+            return null;
+        }
+        return certificate.getPublicKey();
+    }
+
+    /**
+     * Get the certificate for a given key alias from keyStore.
+     *
+     * @param alias the alias name
+     * @return certificate corresponding to the alias
+     * @throws KeyStoreException if the keystore has not been initialized
+     *                           (loaded), or this operation fails for some other reason
+     */
+    public Certificate getCertificate(String alias) throws KeyStoreException {
+        try {
+            return keyStore.getCertificate(alias);
+        } catch (java.security.KeyStoreException e) {
+            throw new KeyStoreException("Failed to load certificate: " + alias, e);
+        }
+    }
+
+    /**
+     * Get the default private key of the service.
+     *
+     * @return default private key
+     * @throws KeyStoreException if the keystore has not been initialized
+     *                           (loaded), if the algorithm for recovering the entry cannot be found,
+     *                           Keystore configuration does not contain
+     *                           the information needed to recover the key (e.g. wrong password)
+     */
+    public PrivateKey getDefaultPrivateKey() throws KeyStoreException {
+        ConfigRegistry configRegistry = ConfigRegistry.getInstance();
+        char[] keyStorePassword = configRegistry.getInstanceConfigValue(KEY_STORE_CONFIG, KEY_PASSWORD)
+                .toCharArray();
```

Comment:
Do we need to check configRegistry.getInstanceConfigValue(KEY_STORE_CONFIG, KEY_PASSWORD) != null here?